processing/src/main/java/io/druid/query/extraction/LookupReferencesManager.java

```diff
@@ -91,7 +91,13 @@ public boolean put(String lookupName, final LookupExtractorFactory lookupExtract
       if (!lookupExtractorFactory.start()) {
         throw new ISE("start method returned false for lookup [%s]", lookupName);
       }
-      return (null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory));
+      final boolean noPrior = null == lookupMap.putIfAbsent(lookupName, lookupExtractorFactory);
```

Comment:
this was a bug in the prior code where lookupExtractorFactory would never be closed if the lookupname was already present.